wall/wall.c

```diff
@@ -70,14 +69,15 @@ void main(int argc, char *argv[]) {
 	strncpy(sterm, ttyname(fileno(stderr)), sizeof(sterm));
 	for (i = 1; sterm[i] != '/'; i++);
 	strncpy(sterm, &sterm[(i + 1)], UT_LINESIZE);
```

Comment:
Please use `, sizeof(sterm));`.

Reply:
UT_LINESIZE is actually the sizeof(sterm). By the way, array bound changes also done.

Reply:
> UT_LINESIZE is actually the sizeof(sterm). By the way, array bound changes also done.

Then why is 32 hardcoded?

Reply:
Well, I think UNIX v7 didn't had this macro yet, so the crew that made this program hardcoded it.

Reply:
Then either choose to hardcode it everywhere or use the macro everywhere

Reply:
Yeah, the macro is being used everywhere now.

Reply:
Just a rectification of the before: since ``sterm[]`` first receives the __file path__ of the terminal line, it should be of size ``PATH_MAX`` and then the terminal line itself should be extracted from it (using that for loop) and copied back using ``UT_LINESIZE`` --- that is, the limit that a terminal line have in the utmp(x) database, without the base ``/dev`` path --- as the limit.
I think that's pretty much it.